lib/plugins/aws/lib/naming.js

```diff
@@ -0,0 +1,174 @@
+'use strict';
+
+const _ = require('lodash');
+
+module.exports = {
+  /*
+   * General Name Normalization
+   */
+  normalizeName(name) {
+    return `${_.upperFirst(name)}`;
+  },
+  normalizeNameToAlphaNumericOnly(name) {
+    return this.normalizeName(name.replace(/[^0-9A-Za-z]/g, ''));
+  },
+  normalizePathtoCapitalAlphaNumbericOnlyWithReplacement(path) {
+    return this.normalizeNameToAlphaNumericOnly(
+      path.replace(/-/g, 'Dash')
+        .replace(/\{(.*)\}/g, '$1Var'));
+  },
+
+  getServiceEndpointRegex() {
+    return /^ServiceEndpoint/;
+  },
+
+  /*
+   * Stack Naming
+   */
+  getStackName() {
+    return `${this.sdk.serverless.service.service}-${this.sdk.getStage()}`;
+  },
+
+  getLogicalLogGroupName(functionName) {
+    return `${this.getNormalizedLambdaName(functionName)}LogGroup`;
+  },
+
+  /*
+   * Lambda Function Naming
+   */
+  getNormalizedLambdaName(functionName) {
+    return this.normalizeName(functionName);
+  },
+  extractLambdaNameFromArn(functionArn) {
+    return functionArn.substring(functionArn.lastIndexOf(':') + 1);
+  },
+  extractAuthorizerNameFromArn(functionArn) {
+    const splitArn = functionArn.split(':');
+    // TODO the following two lines assumes default function naming?  Is there a better way?
+    // TODO (see ~/lib/classes/Service.js:~155)
```

Comment:
These are more of a general architectural issues that we need to discuss brought up by Erik. I kept it for future reference along with other TODOs